externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteScala.scala

```diff
@@ -64,8 +68,19 @@ class ExecuteScala(
       if (legacyOutput.nonEmpty) {
         warn(s"Clearing legacy output from last interpreting:\n $legacyOutput")
       }
-      val jars = spark.sharedState.jarClassLoader.getURLs
-      repl.addUrlsToClassPath(jars: _*)
+      spark.sharedState.jarClassLoader.getURLs.foreach { jar =>
+        try {
+          if ("file".equals(jar.toURI.getScheme)) {
+            repl.addUrlsToClassPath(jar)
+          } else {
+            spark.sparkContext.addFile(jar.toString)
```

Comment:
it is tricky, but I can not find another solution for this.

cc @yaooqinn  @iodone @942011334

Reply:
`SparkSession.sessionState.resourceLoader.addJar` : 
![image](https://user-images.githubusercontent.com/5451385/185852694-297785e0-78e5-4401-b235-4eedc28ee065.png)
There is a difference between the implementation of SparkContext.addJar` and `SparkSession.sessionState.resourceLoader.addJar`. `sparkcontext.add jar` does not have a jar path to add to the `classpath` of the current thread context

Reply:
Why do we need to call `sparkContext.addjar` repeatedly here?

Reply:
the jar url is in remote FS and scala repl can not identify that.

Reply:
the jar url is in remote FS and scala repl can not identify that.